angular-app/src/app/project.service.ts

```diff
@@ -29,7 +42,28 @@ export class ProjectService {
   }
 
   public getProject(title: string): Observable<Project> {
-    console.log(title);
-    return this.firestore.collection("projects").doc<Project>(title).valueChanges();
+    return this.firestore.collection(COLLECTION_PROJECTS).doc<Project>(title).valueChanges();
+  }
+
+  public getProjectFiles(title: string): Observable<ProjectFile[]> {
+    return this.firestore.collection(COLLECTION_PROJECTS).doc<Project>(title).collection<ProjectFile>(COLLECTION_FILES).valueChanges();
+  }
+
+  public addProject(project: Project): void {
+    this.firestore.collection<Project>(COLLECTION_PROJECTS).doc(project.title).set(project)
+    .catch(error => console.error("Error adding document: ", error));
+  }
+
+  public uploadFile(blobstoreUploadUrl: string, project: Project, file: File): void {
+    const formData: FormData = new FormData();
+    formData.append('project', project.title);
+    formData.append('filename', file.name);
+    formData.append('file', file, file.name);
+    this.httpClient
+    .post(blobstoreUploadUrl, formData)
```

Comment:
Is the blobstoreUpload where the media files will go?

Reply:
The text document can probably be saved directly into Firestore